src/unix/bsd/mod.rs

```diff
@@ -388,6 +389,10 @@ extern {
     pub fn getpwent() -> *mut passwd;
     pub fn setpwent();
     pub fn endpwent();
+    pub fn setgrent();
```

Comment:
setgrent is not available in freebsd (or it looks like this is the case), you are going to have to put it in the folders of all bsd targets except for the freebsd one :/

Reply:
setgrent is available in FreeBSD. What gives you the impression it isn't?

https://github.com/freebsd/freebsd/blob/d5842c73935af7f7a7eeaeaec75ad8f3cf6cdf50/include/grp.h#L80

Reply:
> setgrent is available in FreeBSD. What gives you the impression it isn't?

Indeed, this buildbot: https://travis-ci.org/rust-lang/libc/jobs/347737123#L1402

Says that the pointer types are incompatible. It looks like its signature was changed in: https://github.com/freebsd/freebsd/commit/786c3926a65879b5dd6c301ca9ccdfda486fd195

from `int setgrent()` to `void setgrent()`. That commit was 2 years ago, but it might be that we are testing against a FreeBSD release that is older as the release where that commit was included. 

You might want to change the signature to `int setgrent()` and see if that fixes the build. If so, please open an issue about updating the FreeBSD version (we can fix that and fix setgrent later).

Reply:
If it turns out to be the case that changing the signature to `int setgrent()` fixes this issue, what you can also do is change it back afterwards to `void setgrent()` and whitelist the function here:  https://github.com/rust-lang/libc/blob/master/libc-test/build.rs#L606

By adding a match arm like this:

```rust
// FIXME: need to upgrade FreeBSD version. Issue: rust-lang/libc/xxxxxx
"setgrent" if freebsd => true,
```